langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Sort.java

```diff
@@ -42,12 +48,16 @@
 public class Sort {
 
     public static BArray sort(BArray arr, Object direction, Object func) {
-        checkIsArrayOnlyOperation(TypeUtils.getImpliedType(arr.getType()), "sort()");
+        BArray sortedArray;
+        Type arrType = arr.getType();
         BFunctionPointer<Object, Object> function = (BFunctionPointer<Object, Object>) func;
-
+        // Check if the array type is an Ordered type, otherwise a key function is mandatory
+        if (!isOrderedType(arrType) && function == null) {
+            throw ErrorCreator.createError(getModulePrefixedReason(ARRAY_LANG_LIB, INVALID_TYPE_TO_SORT),
+                    StringUtils.fromString("Valid key function required"));
```

Comment:
Is this error message describe the real issue in here

Reply:
The array or tuple does not need a key function if it's type is an `OrderedType`. Otherwise, the sort operation is still supported only if a valid key function is provided with it. The error message is displayed if the array/tuple type is **not** an `OrderedType` and a valid key function is **not** provided. IMO the issue is the lack of a key function. When this happens, I want the user to provide a valid key function instead of trying to change the type. Can I improve the error message in any way?